protocol/x/vault/types/params_test.go

```diff
@@ -88,3 +89,54 @@ func TestValidateQuotingParams(t *testing.T) {
 		})
 	}
 }
+
+func TestValidateVaultParams(t *testing.T) {
+	tests := map[string]struct {
+		// Params to validate.
+		params types.VaultParams
+		// Expected error
+		expectedErr error
+	}{
+		"Success": {
+			params: types.VaultParams{
+				Status:        types.VaultStatus_VAULT_STATUS_QUOTING,
+				QuotingParams: &constants.QuotingParams,
+			},
+			expectedErr: nil,
+		},
+		"Success - nil quoting params": {
+			params: types.VaultParams{
+				Status: types.VaultStatus_VAULT_STATUS_QUOTING,
+			},
+			expectedErr: nil,
+		},
+		"Failure - unspecified vault status": {
+			params: types.VaultParams{
+				QuotingParams: &constants.QuotingParams,
+			},
+			expectedErr: types.ErrUnspecifiedVaultStatus,
+		},
+		"Failure - invalid quoting params": {
+			params: types.VaultParams{
+				Status: types.VaultStatus_VAULT_STATUS_QUOTING,
+				QuotingParams: &types.QuotingParams{
```

Comment:
nit: Can you re-use the constant that you declared for invalid quoting params here?